src/components/widget/DraggableWidget.tsx

```diff
@@ -4,40 +4,71 @@ import styles from './Widget.scss';
 import {isInverted} from './utils';
 import {DragTracker, IVec2} from 'utils/move';
 import {IWidgetState, IWidgetProps, default as Widget} from './Widget';
+import {AnnotationAPIModelAttrs} from 'api/annotations';
+import {AnnotationPriorities} from '../consts';
+import * as _ from 'lodash';
+import {IEditorProps, IEditorState} from 'components/editor/interfaces';
 
-export interface IDraggableWidgetProps extends IWidgetProps {
+export interface IDraggableWidgetProps {
+  // Props consumed by DraggableWidget
   mover: RefObject<HTMLElement>;
-  onDrag: (delta: IVec2) => void;
+  initialLocationX: number;
+  initialLocationY: number;
+
+  // Props passed to Widget
+  children: any;
+  widgetTriangle: boolean;
+  className: string;
+}
+
+export interface IDraggableWidgetState {
+  initialLocationX: number;
+  initialLocationY: number;
+  locationX: number;
+  locationY: number;
+  hasBeenDragged: boolean;
 }
 
 export default class DraggableWidget extends React.PureComponent<
     Partial<IDraggableWidgetProps>,
-  {}
+    Partial<IDraggableWidgetState>
   > {
 
   static defaultProps = {
-    ...Widget.defaultProps,
+    initialLocationX: 0,
+    initialLocationY: 0,
   };
 
-  static getDerivedStateFromProps(nextProps) {
-    return {
-      locationX: nextProps.locationX,
-      locationY: nextProps.locationY,
-      moved: false,
-    };
+  static getDerivedStateFromProps(nextProps: IDraggableWidgetProps, prevState: IDraggableWidgetState) {
+    const areInitialLocationsEqual =
+      prevState.initialLocationX === nextProps.initialLocationX
+      && prevState.initialLocationY === nextProps.initialLocationY;
+    if (areInitialLocationsEqual) {
```

Comment:
This simplified equivalent (only non null returned value changes state):
```
if (!areInitialLocationsEqual) {
 return {
        initialLocationX: nextProps.initialLocationX,
        initialLocationY: nextProps.initialLocationY,
        locationX: nextProps.initialLocationX,
        locationY: nextProps.initialLocationY,
        hasBeenDragged: false,
      };
}
```

Reply:
OK, that's a good point, still I would use that with one extra `return null` line :)

Reply:
Sure, that's what I meant :)